src/pull_request.py

```diff
@@ -0,0 +1,230 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Module for handling git repository."""
+
+import logging
+import re
+import typing
+from uuid import uuid4
+
+from git import GitCommandError
+from git.repo import Repo
+from github import Github
+from github.Repository import Repository
+
+from .exceptions import GitError, InputError
+
+GITHUB_HOSTNAME = "github.com"
+HTTPS_URL_PATTERN = re.compile(rf"^https?:\/\/.*@?{GITHUB_HOSTNAME}\/(.+\/.+?)(.git)?$")
+ACTIONS_USER_NAME = "actions-bot"
+ACTIONS_USER_EMAIL = "actions-bot@users.noreply.github.com"
+ACTIONS_COMMIT_MESSAGE = "migrate docs from server"
+ACTIONS_PULL_REQUEST_TITLE = "[docs] Migrate charm docs"
+ACTIONS_PULL_REQUEST_BODY = "This pull request was autogenerated by upload-charm-docs"
+PR_LINK_NO_CHANGE = "<not created due to no changes in repository>"
+PR_LINK_DRY_RUN = "<not created due in dry run mode>"
+DEFAULT_BRANCH_NAME = "upload-charm-docs"
+
+
+def _configure_user(repository: Repo):
```

Comment:
It would be nice to add the return types to these functions

Reply:
Added, thank you for the feedback!